bedrock/firefox/views.py

```diff
@@ -220,9 +220,13 @@ def send_to_device_ajax(request):
 
                 try:
                     basket.request('post', 'subscribe_sms', data=data)
-                except basket.BasketException:
-                    return HttpResponseJSON({'success': False, 'errors': ['system']},
-                                            status=400)
+                except basket.BasketException as e:
+                    if e.desc == 'mobile_number is invalid':
+                        return HttpResponseJSON({'success': False, 'errors': ['number']},
+                                                status=400)
```

Comment:
Thinking we should change this to `status=200` so we continue to show the "Sorry, we can’t send SMS messages to this phone number. " error message. As is, an invalid number displays the "An error occurred in our system. Please try again later." message, which is really accurate (or helpful).

Reply:
Oh, so the JS will interpret a 400 response as an "error" and send it though the error callback which won't display the "number" message?

Reply:
I've fixed this so that the view and test use a 200 response so that the JS will interpret the error codes.